cilium/cmd/service_update.go

```diff
@@ -148,15 +150,19 @@ func updateService(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	if len(backendWeights) == 0 {
```

Comment:
I suggest to also check whether user specified all backend weights, otherwise we might run into index out of range below
```suggestion
	if len(backendWeights) == 0 || len(backendWeights) != len(backends) {
```